.github/workflows/bindists.yaml

```diff
@@ -48,6 +48,9 @@ jobs:
           - image: debian:11
             installCmd: apt-get update && apt-get install -y
             toolRequirements: build-essential curl libffi-dev  libgmp-dev libgmp10 libncurses-dev libncurses5 libtinfo5 libnuma-dev
+          - image: debian:unstable
+            installCmd: apt-get update && apt-get install -y
+            toolRequirements: build-essential curl libffi-dev  libgmp-dev libgmp10 libncurses-dev libncurses6 libtinfo6 libnuma-dev
```

Comment:
This is actually a valid configuration for the `debian:12` image. If you take a look here https://packages.debian.org/source/stable/ncurses, you will see that `libncurses5` and `libtinfo5` are just legacy packages. The `libncurses-dev` package depends on the new ones:
```shell
# apt show libncurses-dev 2>&1 | grep ^Depends
Depends: libtinfo6 (= 6.4-4), libncurses6 (= 6.4-4), libncursesw6 (= 6.4-4), libc6-dev | libc-dev, ncurses-bin (>= 6.0+20151017)
```

I would actually do an other thing: I would completely remove the binary packages from those commands for all the Debian/Ubuntu releases. In those distros every `-dev` package depends on the corresponding binary library package. So, instead of writing:
```
toolRequirements: build-essential curl libffi-dev  libgmp-dev libgmp10 libncurses-dev libncurses6 libtinfo6 libnuma-dev
```
We can just write:
```
toolRequirements: build-essential curl libffi-dev  libgmp-dev libncurses-dev libnuma-dev
```
Btw, we are already using this approach with the numa library, we just require libnuma-dev, and `apt` will install `libnuma1` too.

The only difference would be that the packages for the library binaries (eg: `libncurses6`), will not be marked as automatically installed, and this is not really important, because the binaries and the headers are both required.

We could have something like this:
```yaml
          - image: debian:10
            installCmd: apt-get update && apt-get install -y
            toolRequirements: build-essential curl libffi-dev libgmp-dev libncurses-dev libnuma-dev
          - image: debian:11
            installCmd: apt-get update && apt-get install -y
            toolRequirements: build-essential curl libffi-dev libgmp-dev libncurses-dev libnuma-dev
          - image: debian:12
            installCmd: apt-get update && apt-get install -y
            toolRequirements: build-essential curl libffi-dev libgmp-dev libncurses-dev libnuma-dev
```

In this way we will not have to never touch the dependencies :)

I'm not sure it's really useful to have the image for debian unstable. It will change during time, and the ghc packages built on it will be invalid in the medium/long term.
I think we should add `debian:12`, and as I suggested in the other PR, we should use the most update version with _unknown_ versions.

Reply:
Ah, we should also fix the documentation available here https://www.haskell.org/ghcup/install/#version-12.
I'm pretty in favor of just suggesting `build-essential curl libffi-dev libgmp-dev libncurses-dev libnuma-dev`, but if you think its better to be more explicit, we should replace `libncurses5` and `libtinfo5` with the soname 6 versions, and we should also mention `libnuma1` and `libnuma-dev`.

Reply:
> This is actually a valid configuration for the debian:12 image.

Then we add a separate section for debian:12.

> I would actually do an other thing: I would completely remove the binary packages from those commands for all the Debian/Ubuntu releases. In those distros every -dev package depends on the corresponding binary library package. So, instead of writing:

This is not correct as explained below.

Reply:
I'm pretty sure is correct :) See below